keel-core/src/main/kotlin/com/netflix/spinnaker/keel/dryrun/ChangeSummary.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.keel.dryrun
+
+import com.netflix.spinnaker.keel.state.FieldState
+
+open class ChangeSummary {
+  var type: ChangeType = ChangeType.NO_CHANGE
+  var summary: MutableList<String> = mutableListOf()
+  var diff: List<FieldState> = emptyList()
+
+  fun addMessage(msg: String) {
+    summary.add(msg)
+  }
+
+  override fun toString(): String {
+    // TODO eb: more readable friendly format for logs/response?
+    var msg = "ChangeSummary(type=$type, summary=$summary"
+    if (!listOf(ChangeType.CREATE, ChangeType.NO_CHANGE).contains(type)) msg += ", diff=$diff"
+    msg += ")"
+    return msg
+  }
+
+
+}
```

Comment:
A couple things here:

1. Don't think we need this to be `open`.
2. Why all `var`? Seems to me like we could switch this to being a `data class` and have the properties immutable? If someone needs to change it, they could just perform a copy.

Reply:
I'll change to data class. But, the reason they're var is so this object can have defaults but be built/updated as we know what's changing. Would you prefer to create a new object each time instead of updating fields on this one?